api/controller/journal.js

```diff
@@ -1221,3 +1221,57 @@ exports.findAllEntries = function(req, res) {
 		});
 	}
 };
+
+exports.copyEntry = function (entryDoc) {
+	return new Promise(function (resolve, reject) {
+		if (typeof entryDoc == "undefined") {
+			reject("Entry is undefined");
+		}
+		if (!typeof entryDoc.text == "string" || mongo.ObjectID.isValid(entryDoc.text)) {
```

Comment:
it checks that if text is not a  `"string"`  and  text is a valid `objectId`, so we do copy that entry, 
and now i feel that you are right , this can be work with ` mongo.ObjectID.isValid(entryDoc.text)` this only !